src/textual/layouts/vertical.py

```diff
@@ -40,14 +38,11 @@ def arrange(
 
         y = box_models[0].margin.top if box_models else 0
 
-        displayed_children = parent.displayed_children
-
+        displayed_children = cast("list[Widget]", parent.displayed_children)
         for widget, box_model, margin in zip(displayed_children, box_models, margins):
             content_width, content_height = box_model.size
-            offset_x = widget.styles.align_width(content_width, parent_size.width)
+            offset_x = widget.styles.align_width(content_width, size.width)
```

Comment:
THAT'S IT, THAT'S THE BUGFIX! :sweat_smile: 
+482 lines of code in this PR's diff, just for this one-line bugfix ^^